plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizardModel.java

```diff
@@ -117,7 +121,7 @@ private IApplicationSource getLocalAppSource(String filename) {
 		IResource resource = null;
 		filename = VariablesHelper.replaceVariables(filename);
 		try {
-			if (!Files.isRegularFile(Paths.get(filename))) {
+			if (!OpenshiftUIConstants.URL_VALIDATOR.isValid(filename) && !Files.isRegularFile(Paths.get(filename))) {
 				return null;
 			}
 			resource = resourceFactory.create(createInputStream(filename));
```

Comment:
Following on last comment, we should check the inputstream is empty before trying to create the resource. That'll give is more info on debugging.
```java 
String content = IOUtils.toString(is);
if (StringUtils.isBlank(content)) {
  throw new OpenShiftException(NLS.bind("\"{0}\" is empty", filename));
}
resource = resourceFactory.create(content);
```